XamarinCommunityToolkit/XamarinCommunityToolkit.csproj

```diff
@@ -1,71 +1,70 @@
+    <AssemblyName>XamarinCommunityToolkit</AssemblyName>
+    <RootNamespace>XamarinCommunityToolkit</RootNamespace>
+    <PackageId>XamarinCommunityToolkit</PackageId>
+    <PackageIconUrl>
+    </PackageIconUrl>
+    <Summary>Community-created toolkit with common Xamarin converters, effects, behaviors etc.</Summary>
+    <PackageTag>xamarin,ios,android,uwp,xamarin.forms,effects,controls,converters,animations,toolkit,kit,communitytoolkit,xamarincommunitytoolkit,watchos,tvos,tizen</PackageTag>
+    <Title>XamarinCommunityToolkit</Title>
+    <Description>The Xamarin Community Toolkit is a collection of Animations, Behaviors, Converters, and Effects for mobile development with Xamarin.Forms. It simplifies and demonstrates common developer tasks building iOS, Android, and UWP apps with Xamarin.Forms.</Description>
+    <Product>$(AssemblyName) ($(TargetFramework))</Product>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <AssemblyFileVersion>1.0.0.0</AssemblyFileVersion>
+    <Version>1.0.0</Version>
+    <PackageVersion>$(Version)$(VersionSuffix)</PackageVersion>
+    <Authors>Microsoft</Authors>
+    <Owners>microsoft</Owners>
+    <NeutralLanguage>en</NeutralLanguage>
+    <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
+    <RepositoryUrl>https://github.com/xamarin/XamarinCommunityToolkit</RepositoryUrl>
+    <PackageReleaseNotes>
+      <!-- See if we can host these on MS docs? -->
+    </PackageReleaseNotes>
+    <DefineConstants>$(DefineConstants);</DefineConstants>
+    <UseFullSemVerForNuGet>false</UseFullSemVerForNuGet>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <PackageProjectUrl>https://github.com/xamarin/XamarinCommunityToolkit</PackageProjectUrl>
+    <DebugType>portable</DebugType>
+    <Configurations>Debug;Release</Configurations>
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)'=='Debug' ">
+    <DebugSymbols>true</DebugSymbols>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)'=='Release' And '$(OS)' == 'Windows_NT' ">
+    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <!-- Embed source files that are not tracked by the source control manager in the PDB -->
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+    <!-- Build symbol package (.snupkg) to distribute the PDB containing Source Link -->
+    <IncludeSymbols>true</IncludeSymbols>
+    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
+  </PropertyGroup>
+  <ItemGroup Condition=" '$(Configuration)'=='Release' ">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="**/*.cs" Exclude="obj\**" />    
```

Comment:
So this pattern represents somewhat of a standard when creating a multi-targeting library. The conditional compilation statements pick up on specific files that are written for a specific platform and only include those when compiling for said platform. 

If I have filenames ending with `.ios.cs` it will be iOS specific code. That code won't compile on other platforms. By including `*.cs` it means that all code files (even platform-specific ones) will always be included in each portion of the library, including iOS-specific files on Android, Android files on iOS, etc.

Good examples of this can be found in the Xamarin.Essentials library:
https://github.com/xamarin/Essentials/tree/develop/Xamarin.Essentials/Barometer

As you can see, the generic .NETStandard compliant implementation is in the `.shared.cs` file whereas platform-specific ones are in their own file. For our converters and behaviors, we will probably never write platform-specific code, which is why at the moment this compiles. However, as soon as we will add a platform-specific file, such as in the case of controls and effects,  this will become more prevalent and compilation will fail.

Reply:
Okay, I understand the reason behind it now.  Thank you for pointing me toward a real example so I can see where it's used.  I'll rename the Behaviors to "*.shared.cs" tonight.